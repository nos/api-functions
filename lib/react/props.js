"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.assetProps = exports.nosProps = void 0;

var _propTypes = require("prop-types");

var nosProps = (0, _propTypes.shape)({
  exists: _propTypes.bool.isRequired,
  getAddress: _propTypes.func.isRequired,
  getBalance: _propTypes.func.isRequired,
  getLastBlock: _propTypes.func.isRequired,
  claimGas: _propTypes.func.isRequired,
  send: _propTypes.func.isRequired,
  testInvoke: _propTypes.func.isRequired,
  invoke: _propTypes.func.isRequired,
  getStorage: _propTypes.func.isRequired,
  getPublicKey: _propTypes.func.isRequired,
  encrypt: _propTypes.func.isRequired,
  decrypt: _propTypes.func.isRequired
});
exports.nosProps = nosProps;
var assetProps = (0, _propTypes.objectOf)(_propTypes.string);
exports.assetProps = assetProps;