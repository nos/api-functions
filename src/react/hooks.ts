import { assets, nos } from "../index";

export const useNOS = () => nos;

export const useAssets = () => assets;
