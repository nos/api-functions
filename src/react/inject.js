import React from "react";

import { nos, assets } from "../";

export const injectNOS = Component => props => (
  <Component {...props} nos={nos} />
);

export const injectAssets = Component => props => (
  <Component {...props} assets={assets} />
);
