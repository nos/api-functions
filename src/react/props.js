import { func, string, shape, objectOf } from "prop-types";

export const nosProps = shape({
  exists: bool.isRequired,
  getAddress: func.isRequired,
  getBalance: func.isRequired,
  claimGas: func.isRequired,
  send: func.isRequired,
  testInvoke: func.isRequired,
  invoke: func.isRequired,
  getStorage: func.isRequired
});

export const assetProps = objectOf(string);
