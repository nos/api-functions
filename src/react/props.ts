import { bool, func, objectOf, shape, string } from "prop-types";

export const nosProps = shape({
  claimGas: func.isRequired,
  decrypt: func.isRequired,
  encrypt: func.isRequired,
  exists: bool.isRequired,
  getAddress: func.isRequired,
  getBalance: func.isRequired,
  getLastBlock: func.isRequired,
  getPublicKey: func.isRequired,
  getStorage: func.isRequired,
  invoke: func.isRequired,
  send: func.isRequired,
  testInvoke: func.isRequired
});

export const assetProps = objectOf(string);
