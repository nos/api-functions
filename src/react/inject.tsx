import * as React from "react";

import { assets, nos } from "../index";
import ApiFunctions from "../nos";

export const injectNOS = (Component: any) => (props: any) => <Component {...props} nos={nos} />;

export const injectAssets = (Component: any) => (props: any) => (
  <Component {...props} assets={assets} />
);

export const NosFunctions: React.SFC<{}> = ({ children }: { children: any }) => children({ nos });

export const NosAssets: React.SFC<{}> = ({ children }: { children: any }) => children({ assets });
