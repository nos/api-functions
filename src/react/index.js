import { nosProps, assetProps } from "./props";
import { injectNOS, injectAssets } from "./inject";

export { injectNOS, injectAssets, nosProps, assetProps };
