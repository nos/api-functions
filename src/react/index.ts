import { useAssets, useNOS } from "./hooks";
import { injectAssets, injectNOS, NosAssets, NosFunctions } from "./inject";
import { assetProps, nosProps } from "./props";

export {
  assetProps,
  injectAssets,
  injectNOS,
  NosAssets,
  NosFunctions,
  nosProps,
  useAssets,
  useNOS
};
