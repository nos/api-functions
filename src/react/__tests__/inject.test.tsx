import { mount } from "enzyme";
import * as React from "react";

import { injectAssets, injectNOS, NosAssets, NosFunctions } from "../inject";

describe("inject", () => {
  it("should add nos to the wrapped component's props", () => {
    const MyComponent = () => <div />;
    MyComponent.displayName = "MyComponent";

    const MyWrappedComponent = injectNOS(MyComponent);

    const wrapper = mount(<MyWrappedComponent />);
    expect(wrapper).toMatchSnapshot();
    expect(Object.keys(wrapper.find("MyComponent").props())).toEqual(["nos"]);
  });

  it("should add assets to the wrapped component's props", () => {
    const MyComponent = () => <div />;
    MyComponent.displayName = "MyComponent";

    const MyWrappedComponent = injectAssets(MyComponent);

    const wrapper = mount(<MyWrappedComponent />);
    expect(wrapper).toMatchSnapshot();
    expect(Object.keys(wrapper.find("MyComponent").props())).toEqual(["assets"]);
  });
});

describe("NosX", () => {
  it("should add nos via render props", () => {
    const MyComponent = () => <div />;
    MyComponent.displayName = "MyComponent";

    const MyWrappedComponent = () => (
      <NosFunctions>{(props: any) => <MyComponent {...props} />}</NosFunctions>
    );

    const wrapper = mount(<MyWrappedComponent />);
    expect(wrapper).toMatchSnapshot();
    expect(Object.keys(wrapper.find("MyComponent").props())).toEqual(["nos"]);
  });

  it("should add assets to the wrapped component's props", () => {
    const MyComponent = () => <div />;
    MyComponent.displayName = "MyComponent";

    const MyWrappedComponent = () => (
      <NosAssets>{(props: any) => <MyComponent {...props} />}</NosAssets>
    );

    const wrapper = mount(<MyWrappedComponent />);
    expect(wrapper).toMatchSnapshot();
    expect(Object.keys(wrapper.find("MyComponent").props())).toEqual(["assets"]);
  });
});
