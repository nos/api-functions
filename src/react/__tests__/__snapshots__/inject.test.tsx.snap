// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NosX should add assets to the wrapped component's props 1`] = `
<MyWrappedComponent>
  <Component>
    <MyComponent
      assets={
        Object {
          "GAS": "602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7",
          "NEO": "c56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b",
        }
      }
    >
      <div />
    </MyComponent>
  </Component>
</MyWrappedComponent>
`;

exports[`NosX should add nos via render props 1`] = `
<MyWrappedComponent>
  <Component>
    <MyComponent
      nos={
        Object {
          "claimGas": [Function],
          "decrypt": [Function],
          "encrypt": [Function],
          "exists": false,
          "getAddress": [Function],
          "getBalance": [Function],
          "getLastBlock": [Function],
          "getPublicKey": [Function],
          "getStorage": [Function],
          "invoke": [Function],
          "off": undefined,
          "on": undefined,
          "once": undefined,
          "send": [Function],
          "testInvoke": [Function],
        }
      }
    >
      <div />
    </MyComponent>
  </Component>
</MyWrappedComponent>
`;

exports[`inject should add assets to the wrapped component's props 1`] = `
<Component>
  <MyComponent
    assets={
      Object {
        "GAS": "602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7",
        "NEO": "c56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b",
      }
    }
  >
    <div />
  </MyComponent>
</Component>
`;

exports[`inject should add nos to the wrapped component's props 1`] = `
<Component>
  <MyComponent
    nos={
      Object {
        "claimGas": [Function],
        "decrypt": [Function],
        "encrypt": [Function],
        "exists": false,
        "getAddress": [Function],
        "getBalance": [Function],
        "getLastBlock": [Function],
        "getPublicKey": [Function],
        "getStorage": [Function],
        "invoke": [Function],
        "off": undefined,
        "on": undefined,
        "once": undefined,
        "send": [Function],
        "testInvoke": [Function],
      }
    }
  >
    <div />
  </MyComponent>
</Component>
`;
